asicdump/asicdump.cpp

```diff
@@ -0,0 +1,291 @@
+    std::cout << "    -h --help:" << std::endl;
+    std::cout << "        Print out this message" << std::endl;
+}
+
+CmdOptions handleCmdLine(int argc, char **argv)
+{
+    SWSS_LOG_ENTER();
+
+    CmdOptions options;
+
+    options.shortNames = false;
+    options.disableColors = false;
+
+    const char* const optstring = "Csh";
+
+    while(true)
+    {
+        static struct option long_options[] =
+        {
+            { "disableColors",  no_argument,       0, 'C' },
+            { "shortNames",     no_argument,       0, 's' },
+            { "help",           no_argument,       0, 'h' },
+            { 0,                0,                 0,  0  }
+        };
+
+        int option_index = 0;
+
+        int c = getopt_long(argc, argv, optstring, long_options, &option_index);
+
+        if (c == -1)
+        {
+            break;
+        }
+
+        switch (c)
+        {
+            case 'C':
+                SWSS_LOG_NOTICE("Disable colors");
+                options.disableColors = true;
+                break;
+
+            case 's':
+                SWSS_LOG_NOTICE("Short names");
+                options.shortNames = true;
+                break;
+
+            case 'h':
+                printUsage();
+                exit(EXIT_SUCCESS);
+
+            case '?':
+                SWSS_LOG_WARN("unknown option %c", optopt);
+                printUsage();
+                exit(EXIT_FAILURE);
+
+            default:
+                SWSS_LOG_ERROR("getopt_long failure");
+                exit(EXIT_FAILURE);
+        }
+    }
+
+    return options;
+}
+
+size_t get_max_attr_len(const TableMap& map)
+{
+    SWSS_LOG_ENTER();
+
+    size_t max = 0;
+
+    for (const auto&field: map)
+    {
+        max = std::max(max, field.first.length());
+    }
+
+    return max;
+}
+
+std::string pad_string(std::string s, size_t pad)
+{
+    SWSS_LOG_ENTER();
+
+    size_t len = s.length();
+
+    if (len < pad)
+    {
+        s.insert(len, pad - len, ' ');
+    }
+
+    return s;
+}
+
+const TableMap* get_table_map(sai_object_id_t object_id)
+{
+    SWSS_LOG_ENTER();
+
+    auto it = g_oid_map.find(object_id);
+
+    if (it == g_oid_map.end())
+    {
+        SWSS_LOG_ERROR("unable to find oid %llx in oid map", object_id);
+        throw;
+    }
+
+    return it->second;
+}
+
+void print_attributes(size_t indent, const TableMap& map)
+{
+    SWSS_LOG_ENTER();
+
+    size_t max_len = get_max_attr_len(map);
+
+    std::string str_indent = pad_string("", indent);
+
+    for (const auto&field: map)
+    {
+        const sai_attr_metadata_t *meta;
+        sai_deserialize_attr_id(field.first, &meta);
+
+        std::stringstream ss;
+
+        ss << str_indent << pad_string(field.first, max_len) << " : ";
+
+        ss << field.second;
+
+        std::cout << ss.str() << std::endl;
+    }
+}
+
+int main(int argc, char ** argv)
+{
+    swss::Logger::getInstance().setMinPrio(swss::Logger::SWSS_DEBUG);
```

Comment:
i want log enter to be logged in main to know when new process started, i will add leter on other notice enter